coalib/output/Logging.py

```diff
@@ -1,10 +1,26 @@
 from datetime import datetime
+from collections import defaultdict
 import json
 import io
 import logging
 import logging.config
 
 
+class CounterHandler(logging.Handler):
+    """
+    A logging handler which counts the number of calls
+    for each logging level.
+    """
+    level_called = None
+
+    def __init__(self, *args, **kwargs):
+        super(CounterHandler, self).__init__(*args, **kwargs)
+        CounterHandler.level_called = defaultdict(lambda: 0)
```

Comment:
@NiklasMM U could also define a `pytest.fixture` in a **tests/conftest.py** that resets `level_called` for every test where it's needed:

```python
@pytest.fixture(scope='function')
def reset_logging_counter(request):
    CounterHandler.level_called = defaultdict(lambda: 0)
    return CounterHandler
```

Then you just add a `reset_logging_counter` argument to every `test_...` method where the dict shall be reset:

```python
def test_...(self, reset_logging_counter):
    ...
```

And you can then also perform additional asserts on the `reset_logging_counter` class itself, like checking if the counting really works correctly for all levels...

It would also be generally good to define an explicit `reset` method:

```python
class CounterHandler(...):
    ...
    @classmethod
    def reset(cls):
        cls.level_called = defaultdict(lambda: 0)
    ...

CounterHandler.reset()
```

Then you can also use `reset` in the fixture and not reimplement logic there which has actually to be tested :)

```python
@pytest.fixture(scope='function')
def reset_logging_counter(request):
    CounterHandler.reset()
    return CounterHandler
```

You can read more about fixtures here: http://doc.pytest.org/en/latest/fixture.html

I still think it's not good when every instantiation of `CounterHandler` resets the counters.

**CC** @jayvdb @sils